src/components/MessagesList/MessagesGroup/Message/MessagePart/Reactions.vue

```diff
@@ -0,0 +1,246 @@
+-->
+
+<template>
+	<!-- reactions buttons and popover with details -->
+	<div class="reactions-wrapper">
+		<NcPopover v-for="reaction in Object.keys(detailedReactions ?? plainReactions)"
+			:key="reaction"
+			:delay="200"
+			:focus-trap="false"
+			:triggers="['hover']"
+			@after-show="fetchReactions">
+			<template #trigger>
+				<NcButton :type="userHasReacted(reaction) ? 'primary' : 'secondary'"
+					class="reaction-button"
+					@click="handleReactionClick(reaction)">
+					{{ reaction }} {{ reactionsCount(reaction) }}
+				</NcButton>
+			</template>
+
+			<div v-if="hasReactions" class="reaction-details">
+				<span>{{ getReactionSummary(reaction) }}</span>
+			</div>
+		</NcPopover>
+
+		<!-- More reactions picker -->
+		<NcEmojiPicker v-if="canReact && hasReactions"
+			:per-line="5"
+			:container="`#message_${id}`"
+			@select="handleReactionClick"
+			@after-show="emitEmojiPickerStatus"
+			@after-hide="emitEmojiPickerStatus">
+			<NcButton class="reaction-button"
+				:aria-label="t('spreed', 'Add more reactions')">
+				<template #icon>
+					<EmoticonOutline :size="15" />
+				</template>
+			</NcButton>
+		</NcEmojiPicker>
+	</div>
+</template>
+
+<script>
+import EmoticonOutline from 'vue-material-design-icons/EmoticonOutline.vue'
+
+import { showError } from '@nextcloud/dialogs'
+
+import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
+import NcEmojiPicker from '@nextcloud/vue/dist/Components/NcEmojiPicker.js'
+import NcPopover from '@nextcloud/vue/dist/Components/NcPopover.js'
+
+import { ATTENDEE } from '../../../../../constants.js'
+import { useGuestNameStore } from '../../../../../stores/guestName.js'
+import { useReactionsStore } from '../../../../../stores/reactions.js'
+
+export default {
+	name: 'Reactions',
+
+	components: {
+		NcButton,
+		NcEmojiPicker,
+		NcPopover,
+		EmoticonOutline,
+	},
+
+	props: {
+		token: {
+			type: String,
+			required: true,
+		},
+		/**
+		 * Whether the current user can react to the message.
+		 */
+		canReact: {
+			type: Boolean,
+			default: false,
+		},
+		/**
+		 * The message id.
+		 */
+		id: {
+			type: [String, Number],
+			required: true,
+		},
+	},
+
+	emits: ['emoji-picker-toggled'],
+
+	setup() {
+		const guestNameStore = useGuestNameStore()
+		const reactionsStore = useReactionsStore()
+		return {
+			guestNameStore,
+			reactionsStore,
+		 }
+	},
+
+	data() {
+		return {
+			detailedReactionsLoading: false,
+		}
+	},
+
+	computed: {
+		hasReactions() {
+			return Object.keys(Object(this.detailedReactions)).length !== 0
+		},
+
+		detailedReactions() {
+			return this.reactionsStore.getReactions(this.token, this.id)
+		},
+
+		plainReactions() {
+			return this.$store.getters.message(this.token, this.id).reactions
+		},
+
+		reactionsSelf() {
+			return this.$store.getters.message(this.token, this.id).reactionsSelf
+		},
+
+		actorId() {
+			return this.$store.getters.getActorId()
+		},
+	},
+
+	methods: {
+		fetchReactions() {
```

Comment:
Not covered by tests, try https://test-utils.vuejs.org/api/#trigger

Also might be an edge case, but if somehow there is a misalignment `detailedReactions.length != plainReactions`, we should also fetch details